sync_diff_inspector/report.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"sync"
+)
+
+// TableResult saves the check result for every table.
+type TableResult struct {
+	Table       string
+	StructEqual bool
+	DataEqual   bool
+}
+
+// Report saves the check results.
+type Report struct {
+	sync.RWMutex
+
+	Schema         string
+	TableNameEqual bool
+	SourceTables   []string
+	TargetTables   []string
+	PassNum        int32
+	FailedNum      int32
+	TableResults   map[string]*TableResult
+	Pass           bool
+}
+
+// NewReport returns a new Report.
+func NewReport(schema string) *Report {
+	return &Report{
+		Schema:         schema,
+		TableResults:   make(map[string]*TableResult),
+		TableNameEqual: true,
+		Pass:           true,
+	}
+}
+
+// String returns a string of this Report.
+func (r *Report) String() (report string) {
```

Comment:
let fail item display firstly

Reply:
give a display example

Reply:
done

Reply:
If u have time, u can use a template to do it that is easier to modify and maintain

Reply:
do u forget read lock in this function?

Reply:
add read lock done. and I will use template next pr.